src/j1client.js

```diff
@@ -103,7 +114,30 @@ class JupiterOneClient {
       if (res.errors) {
         throw new Error(`JupiterOne returned error(s) for query: '${j1ql}'`);
       }
-      const { data } = res.data.queryV1;
+
+      const deferredUrl = res.data.queryV1.url;
+      let status = JobStatus.IN_PROGRESS;
+      let statusFile;
+      var sleep = function(ms) {
+        return new Promise(function(resolve) {
+          return setTimeout(resolve, ms);
+        });
+      };
+      do {
+        await sleep(200);
+        statusFile = await fetch(deferredUrl).then(res => res.json());
+        status = statusFile.status;
+      } while (status == JobStatus.IN_PROGRESS);
+
+      let result;
+      if (status == JobStatus.COMPLETED) {
+        result = await fetch(statusFile.url).then(res => res.json());
+      } else {
+        // JobStatus.FAILED
+        throw new Error(result.error || "Job failed without an error message.");
```

Comment:
Is `result.error` actually going to be a string?

Reply:
This should have been statusFile.error (result isn't yet populated). Not sure how I can test what the response will actually be... any thoughts @philidem ?

Reply:
I borrowed the logic from jqs repo so I believe it to be sound.